scripts/check-versions.sh

```diff
@@ -0,0 +1,49 @@
+#!/usr/bin/env bash
+# Copyright (c) HashiCorp, Inc.
+# SPDX-License-Identifier: BUSL-1.1
+
+set -e
+
+if [ -z "${VERSION}" ]; then
+  echo "VERSION variable must be set" >&2
+  exit 1
+fi
+
+if [ "${VERSION}" == "0.0.0-dev" ]; then
+  echo "dev build, skipping version checks"
+  exit 0
+fi
```

Comment:
Just thinking out loud, someday we may want to relax this a little to allow any `-dev` builds, for testing images internally without doing a full release.

Reply:
See #396